### DISCUSSION TITLE
创建反向代理时如协议为HTTPS默认启用回源SNI

### DISCUSSION THREAD

**Discussion by unknown-o** [1Panel-dev/1Panel#5541] [Ideas]
### 1Panel 版本

1.10.10

### 请描述您的需求或者改进建议

部分网站回源时会验证SNI，如SNI不匹配可能导致反向代理失败（502）
![image](https://github.com/1Panel-dev/1Panel/assets/51825113/4fffc03e-4e6f-4cd4-ba79-588a5c520cac)

### 请描述你建议的实现方案

创建反向代理配置文件时，如协议为https，默认开启SNI，即在反代配置文件中加入proxy_ssl_server_name on;

**Comment by zhengkunwang223** [comment]
此配置非所有网站都需要 所以默认增加此配置可能不是一个好主意

**Comment by unknown-o** [comment]
> 此配置非所有网站都需要 所以默认增加此配置可能不是一个好主意

是否可以作为高级设置中的开关呢？因为部分网站确实会验证sni

**Reply by zhengkunwang223** [reply]
这种配置我感觉还是手动配置一下

**Comment by bddjr** [comment]
你们都没人发现这个issue被重复很多遍了吗？  

https://github.com/1Panel-dev/1Panel/issues/3461

**Reply by unknown-o** [reply]
不好意思，没注意到。下次会提之前会搜索一下的

**Reply by bddjr** [reply]
我是指这个项目的相关负责人没注意到一些问题被重复很多遍了

**Reply by bddjr** [reply]
这问题解决很容易，然而看看原issue，标签加了又删，加了又删，一拖再拖

**Reply by unknown-o** [reply]
> 这问题解决很容易，然而看看原issue，标签加了又删，加了又删，一拖再拖

现在似乎不准备处理了，可能是很少人用到这个功能吧

**Reply by bddjr** [reply]
什么借口都能找出来，加了SNI难道会导致一些不需要识别SNI的报错？怎么证明？  
> 下一个张小龙，何必是张小龙。

**Reply by zhengkunwang223** [reply]
> 什么借口都能找出来，加了SNI难道会导致一些不需要识别SNI的报错？怎么证明？
> 
> > 下一个张小龙，何必是张小龙。

自签证书配置 proxy_ssl_server_name on 的时候 需要增加一些证书链的配置来防止报错
所以默认给所有网站增加并不是一个好主意  
于此相关 我们给所有的 ssl 增加了 
ssl_stapling on;
ssl_stapling_verify on;
这个校验 导致一部分用户执行 nginx reload 的时候 OCSP 校验失败 ，很多操作直接超时
还有 HSTS 配置，也会导致一些不需要 HSTS 的用户出现问题，这些都可以在 issue 中找到
所以在增加配置的时候 我们是非常谨慎的 
我们已经把 HSTS 做成开关形式
后续会把 OCSP 也作为开关形式
至于 SNI 这个也可以作为开关形式提供

**Reply by bddjr** [reply]
所以应该是默认开，可以关掉。不是什么源站都会用自签证书的

**Reply by unknown-o** [reply]
> > 什么借口都能找出来，加了SNI难道会导致一些不需要识别SNI的报错？怎么证明？
> > > 下一个张小龙，何必是张小龙。
> 
> 自签证书配置 proxy_ssl_server_name on 的时候 需要增加一些证书链的配置来防止报错 所以默认给所有网站增加并不是一个好主意 于此相关 我们给所有的 ssl 增加了 ssl_stapling on; ssl_stapling_verify on; 这个校验 导致一部分用户执行 nginx reload 的时候 OCSP 校验失败 ，很多操作直接超时 还有 HSTS 配置，也会导致一些不需要 HSTS 的用户出现问题，这些都可以在 issue 中找到 所以在增加配置的时候 我们是非常谨慎的 我们已经把 HSTS 做成开关形式 后续会把 OCSP 也作为开关形式 至于 SNI 这个也可以作为开关形式提供

嗯嗯，能理解的，希望后续能作为SNI开关。我看了一下宝塔好像也是默认关，且没有SNI开关
